bootloader/uboot.go

```diff
@@ -111,8 +111,20 @@ func (u *uboot) InstallBootConfig(gadgetDir string, blOpts *Options) error {
 			return err
 		}
 
+		// By default (and historically), we assume uboot is built with
+		// CONFIG_SYS_REDUNDAND_ENVIRONMENT=y
+		useHeaderFlagByte := true
+
+		// if there is a "pattern" boot.sel in the gadget snap, we follow its
+		// lead. If opening it as a uboot env fails in any way we just go with
+		// the default.
+		gadgetEnv, err := ubootenv.OpenWithFlags(filepath.Join(gadgetDir, u.ubootEnvFileName), ubootenv.OpenBestEffort)
+		if err == nil {
+			useHeaderFlagByte = gadgetEnv.HeaderFlagByte()
+		}
```

Comment:
I'd move all of this to a new function called `shouldUseHeaderFlagByte` (or something like that) as the only product of this code is actually `useHeaderFlagByte`. It would maybe make even more sense to create a function called `makeUbootCreateOptions` that orchestrates this?

Reply:
Broke this out into a useHeaderFlagByte function. I toyed with the idea of a function that set up the whole opts struct, but it didn't seem to add anything in terms of readability or reusability at the moment. However if you feel strongly I can go back and change it (should be trivial).